### PR TITLE
Fix #3589: Use java.nio.file.Path instead of VirtualBinaryFile

### LINKED ISSUE #3589: Use java.nio.file.Path in JSEnv API

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

**Comment by gzm0** [comment]
TODO after this (provided it gets accepted)

- [ ] add `JSEnv` test cases for default FS provider paths.
- [ ] get rid of io package.

File: io/jvm/src/main/scala/org/scalajs/io/FileVirtualFiles.scala [hunk 1]
```diff
@@ -12,17 +12,12 @@
 
 package org.scalajs.io
 
-import java.io._
+import java.nio.files._
```
Comment (sjrd) [inline io/jvm/src/main/scala/org/scalajs/io/FileVirtualFiles.scala]: I think this was supposed to be `java.nio.file._`.

**Comment by sjrd** [comment]
`NodeJSEnv.scala` refers to `org.scalajs.io.JSUtils.escapeJS`, which it does not have access to anymore, given that the dependency on `io` was dropped. Should we expose `escapeJS` as a utility helper in the `js-envs` project instead of `io`? It seems we don't use it in the linker project anymore; only in JSEnvs in the TestAdapter.

File: project/Build.scala [hunk 2]
```diff
@@ -776,7 +779,7 @@ object Build {
         baseDirectory.value.getParentFile / "test-common/src/main/scala",
       unmanagedSourceDirectories in Test +=
         baseDirectory.value.getParentFile / "test-common/src/test/scala"
-  ).dependsOn(jsEnvs)
+  ).dependsOn(io, jsEnvs)
```
Comment (sjrd) [inline project/Build.scala]: If we move `escapeJS` to `js-envs`, then this change might not be needed after all.

**Comment by gzm0** [comment]
Updated: I have decided to leave build dependency changes to later when I clean up IO entirely.

**Comment by gzm0** [comment]
A doc link issue. Updated.

**Comment by sjrd** [comment]
Do you still intended to do

> add JSEnv test cases for default FS provider paths.

or is it good to merge for you?

**Comment by gzm0** [comment]
This is good to merge for me. I'll add these two things later.

**Pull request merged**